core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -76,12 +77,20 @@ class SharedPreferenceUtil @Inject constructor(val context: Context) {
     get() {
       val storage = sharedPreferences.getString(PREF_STORAGE, null)
       return when {
-        storage == null -> defaultStorage().also(::putPrefStorage)
-        !File(storage).exists() -> defaultStorage()
+        storage == null -> getActualPath(defaultStorage()).also(::putPrefStorage)
+          .also {
+            putStoragePosition(0)
+          }
```

Comment:
We can inline those two functions inside the `also` operator.

Reply:
done